ray-operator/controllers/ray/rayjob_controller_test.go

```diff
@@ -39,274 +38,207 @@ import (
-				time.Second*15, time.Millisecond*500).Should(Equal(int(replicas)), fmt.Sprintf("workerGroup %v", workerPods.Items))
-			for _, workerPod := range workerPods.Items {
-				workerPod.Status.Phase = corev1.PodRunning
-				// TODO: Check https://github.com/ray-project/kuberay/issues/1736.
-				Expect(k8sClient.Status().Update(ctx, &workerPod)).Should(BeNil())
-			}
+				listResourceFunc(ctx, &workerPods, workerFilterLabels, &client.ListOptions{Namespace: namespace}),
+				time.Second*3, time.Millisecond*500).Should(Equal(int(numWorkerPods)), fmt.Sprintf("workerGroup: %v", workerPods.Items))
 
-			// Update the head Pod to Running and PodReady.
+			// The number of head Pods should be 1.
 			headPods := corev1.PodList{}
-			headFilterLabels := client.MatchingLabels{utils.RayClusterLabelKey: myRayCluster.Name, utils.RayNodeGroupLabelKey: "headgroup"}
-			err := k8sClient.List(ctx, &headPods, headFilterLabels, &client.ListOptions{Namespace: "default"}, client.InNamespace(myRayCluster.Namespace))
-			Expect(err).NotTo(HaveOccurred(), "failed list head pods")
-			Expect(len(headPods.Items)).Should(Equal(1), "My head pod list= %v", headPods.Items)
-			headPod := headPods.Items[0]
-			headPod.Status.Phase = corev1.PodRunning
-			for _, cond := range headPod.Status.Conditions {
-				if cond.Type == corev1.PodReady {
-					cond.Status = corev1.ConditionTrue
-				}
+			headFilterLabels := client.MatchingLabels{utils.RayClusterLabelKey: rayCluster.Name, utils.RayNodeGroupLabelKey: "headgroup"}
+			Eventually(
+				listResourceFunc(ctx, &headPods, headFilterLabels, &client.ListOptions{Namespace: namespace}),
+				time.Second*3, time.Millisecond*500).Should(Equal(1), fmt.Sprintf("head Pod: %v", headPods.Items))
+
+			// Update all Pods, including head and worker Pods, to Running and PodReady.
+			allPods = append(allPods, headPods.Items...)
+			allPods = append(allPods, workerPods.Items...)
+
+			for _, pod := range allPods {
+				pod.Status.Phase = corev1.PodRunning
+				// In envtest, if Pod.Status.Phase is set to running, the Pod's PodReady condition becomes true automatically.
+				// Check https://github.com/ray-project/kuberay/issues/1736 for more details.
+				Expect(k8sClient.Status().Update(ctx, &pod)).Should(BeNil())
 			}
-			Expect(k8sClient.Status().Update(ctx, &headPod)).Should(BeNil())
 
 			// The RayCluster.Status.State should be Ready.
 			Eventually(
-				getClusterState(ctx, "default", myRayCluster.Name),
-				time.Second*15, time.Millisecond*500).Should(Equal(rayv1.Ready))
-		})
-
-		It("Dashboard URL should be set", func() {
-			Eventually(
-				getDashboardURLForRayJob(ctx, myRayJob),
-				time.Second*1, time.Millisecond*500).Should(HavePrefix(myRayJob.Name), "Dashboard URL = %v", myRayJob.Status.DashboardURL)
+				getClusterState(ctx, namespace, rayCluster.Name),
+				time.Second*3, time.Millisecond*500).Should(Equal(rayv1.Ready))
 		})
 
-		It("test cluster selector", func() {
```

Comment:
This test totally doesn't make sense. `ClusterSelector` must be set before the RayJob's creation. If users set the `ClusterSelector` after `DashboardURL` is set, the RayJob may stuck there. Maybe we should consider whether deprecate the `ClusterSelector` feature or not`. Remove `"test cluster selector"`.

Reply:
>  Maybe we should consider whether deprecate the ClusterSelector feature or not.

I never really thought the selector made sense because multiple Ray clusterse can share the same label. Instead this field should just be an object reference or just a name field.:
```
clusterRef:
  name: my-raycluster
```

OR 

```
clusterName: my-raycluster
```

Reply:
It makes sense to me, but I prefer to deemphasize this feature by not mentioning it in the documentation and avoiding updates. To submit Ray jobs to an existing RayCluster, users should use `ray job submit` instead of `ClusterSelector`. We may consider deprecating it or decide to revisit it when we receive sufficient feedback on its use cases from users.